node_monitor/node_monitor.py

```diff
@@ -106,15 +113,15 @@ def broadcast_alerts(self) -> None:
             if preferences['notify_email'] == True:
                 recipients = email_recipients[node_provider_id]
                 self.email_bot.send_emails(recipients, subject, msg)
-            if preferences['notify_slack'] == True:
+            if preferences['notify_slack'] == True and self.slack_bot is not None:
                 channel_name = channels[node_provider_id]['slack_channel_name']
                 self.slack_bot.send_message(channel_name, msg)
-            if preferences['notify_telegram_chat'] == True:
-                # TODO: Not Yet Implemented
-                raise NotImplementedError
-            if preferences['notify_telegram_channel'] == True:
-                # TODO: Not Yet Implemented
-                raise NotImplementedError
+            if preferences['notify_telegram_chat'] == True and self.telegram_bot is not None:
+                chat_id = channels[node_provider_id]['telegram_chat_id']
+                self.telegram_bot.send_message_to_chat(chat_id, msg)
+            if preferences['notify_telegram_channel'] == True and self.telegram_bot is not None:
+                channel_id = channels[node_provider_id]['telegram_channel_id']
+                self.telegram_bot.send_message_to_channel(channel_id, msg)
```

Comment:
I noticed these both send a request to the same URL. Is it possible that we use one function here? It would shrink our codebase.

Also, from a broader view, is it necessary that we allow the user to post both to a channel and a broadcast? Would it be enough to just have a function called `post_telegram_group` that takes either a `channel_id` or a `chat_id`? It would make our data schema and architecture simpler? Do we need both? Just curious.

Reply:
We should probably change our database schema to have separate lookup tables for slack and for telegram at some point, much in the same way we have a table with a `one to many` relationship for emails. We could probably use the same method on all three tables, too.

Reply:
You're right! I was aware of this and initially I wanted to make it explicit even though the code was practically repeated. 

Giving it a bit more thought, indeed a user does not need both a private chat with the telegram-nm-bot AND also a channel.

Thanks for bringing this up!

Reply:
You're right! I was aware of this and initially I wanted to make it explicit even though the code was practically repeated. 

Giving it a bit more thought, indeed a user does not need both a private chat with the telegram-nm-bot AND also a channel.

Thanks for bringing this up!